### PR TITLE
Update CHANGELOG

### PR DESCRIPTION
**Pull request by mernst**
The changes may be easier to read at
https://github.com/smillst/checker-framework/blob/explicit-extends-object-changelog/docs/CHANGELOG.md .

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -16,56 +19,87 @@ Previously, the Checker Framework interpreted
...
+
+As an example, for the Nullness Checker,
+
+ * `<T extends Object>`: change to `<T extends @NonNull Object>`
+ * `<T extends @Nullable Object>`: change to `<T>` for better style
+ * `<T extends @NonNull Object>`: don't change
+ * `<T>`: don't change
+
+ * `<@Nullable T extends Object>`: did not make sense (and has a different meaning now)
+ * `<@Nullable T extends @Nullable Object>`: change to `<T>` for better style
```
Comment (smillst) [inline docs/CHANGELOG.md:75]: This is wrong.  `<T>` is `<@NonNull T extends @Nullable Object>`.
`<@Nullable T extends @Nullable Object>` can be written as just `<@Nullable T>`

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -16,56 +19,87 @@ Previously, the Checker Framework interpreted
...
+   (If the code is annotated for two type systems, then change
+   "<code>&lt;T extends @<em>TopType1 TopType2</em> Object&gt;</code>" to "`<T>`",
+   but leave "<code>&lt;T extends @<em>TopType1 NotTopType2</em> Object&gt;</code>"
+   unchanged.)
+
+For wildcards, the same transformations apply, with one exception.  "`<@Anno ?
+extends @Anno Object>`" can be abbreviated "`<@Anno ?>`", but the same
+abbreviation does not apply to non-wildcard type variables.  That is, "`<@Anno T
+extends @Anno Object>`" cannot be abbreviated unless `@Anno` is the top type, in
+which case it is equivalent to just "`<T>`".
```
Comment (smillst) [inline docs/CHANGELOG.md]: > in which case it is equivalent to just "`<T>`".

This is incorrect:
`<@TopType T>` is **not** equivalent to `<T>`.
`<@TopType T>` is equivalent to `<@TopType T extends @TopType Object>`
`<T>` is equivalent to `<@BottomType T extends @TopType Object>`

File: docs/CHANGELOG.md (hunk 1 above)
Comment (smillst) [inline docs/CHANGELOG.md]: This is wrong.  `<T>` is `<@NonNull T extends @Nullable Object>`.
`<@Nullable T extends @Nullable Object>` can be written as just `<@Nullable T>`

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -16,56 +19,87 @@ Previously, the Checker Framework interpreted
...
+
+ * `<T extends Object>`: change to `<T extends @NonNull Object>`
+ * `<T extends @Nullable Object>`: change to `<T>` for better style
+ * `<T extends @NonNull Object>`: don't change
+ * `<T>`: don't change
+
+ * `<@Nullable T extends Object>`: did not make sense (and has a different meaning now)
+ * `<@Nullable T extends @Nullable Object>`: change to `<T>` for better style
+ * `<@Nullable T extends @NonNull Object>`: did not make sense, and still doesn't
+ * `<@Nullable T>`: change to `<T>` for better style
```
Comment (smillst) [inline docs/CHANGELOG.md]: This is wrong see other comments.

File: docs/CHANGELOG.md [hunk 4]
```diff
@@ -16,56 +19,87 @@ Previously, the Checker Framework interpreted
...
+
+ * `<@NonNull T extends Object>`: change to `<@NonNull T extends @NonNull Object>`
+ * `<@NonNull T extends @Nullable Object>`: change to `<@NonNull T>` for better style
```
Comment (smillst) [inline docs/CHANGELOG.md]: `<@NonNull T extends @Nullable Object>` should be changed to `<T>` 
(`<@NonNull T extends @Nullable Object>`  is also equivalent to  `<@NonNull T>`)

File: docs/CHANGELOG.md [hunk 5]
```diff
@@ -16,56 +19,87 @@ Previously, the Checker Framework interpreted
...
+ * `<@NonNull T extends @NonNull Object>`: don't change
```
Comment (smillst) [inline docs/CHANGELOG.md]: I would write `<@NonNull T extends @NonNull Object>` as `<T extends @NonNull Object>`.

File: docs/CHANGELOG.md [hunk 6]
```diff
@@ -16,56 +19,87 @@ Previously, the Checker Framework interpreted
...
+ * `<@NonNull T>`: don't change
```
Comment (smillst) [inline docs/CHANGELOG.md]: `<@NonNull T>` should be written as `<T>`.

File: docs/CHANGELOG.md [hunk 7]
```diff
@@ -16,56 +19,87 @@ Previously, the Checker Framework interpreted
...
+
+ * `<? extends Object>`: change to `<@NonNull ?>`
+ * `<? extends @Nullable Object>`: change to `<?>` for better style
+ * `<? extends @NonNull Object>`: change to `<@NonNull ?>` for better style
+ * `<?>`: don't change
+
+ * `<@Nullable ? extends Object>`: did not make sense (and has a different meaning now)
+ * `<@Nullable ? extends @Nullable Object>`: change to `<?>` for better style
```
Comment (smillst) [inline docs/CHANGELOG.md]: `<@Nullable ? extends @Nullable Object>`: change to `<?>`.

File: docs/CHANGELOG.md [hunk 8]
```diff
@@ -16,56 +19,87 @@ Previously, the Checker Framework interpreted
...
+ * `<@Nullable ? extends @NonNull Object>`: did not make sense, and still doesn't
+ * `<@Nullable ?>`: change to `<?>` for better style
```
Comment (smillst) [inline docs/CHANGELOG.md]: `<@Nullable ?>` : don't change.

File: docs/CHANGELOG.md [hunk 9]
```diff
@@ -16,56 +19,87 @@ Previously, the Checker Framework interpreted
...
+
+ * `<@NonNull ? extends Object>`: change to `<@NonNull ?>`
+ * `<@NonNull ? extends @Nullable Object>`: change to `<@NonNull ?>` for better style
```
Comment (smillst) [inline docs/CHANGELOG.md]: `<@NonNull ? extends @Nullable Object>`: change to `<?>`

**Review by smillst — CHANGES_REQUESTED** [review]

**Comment by mernst** [comment]
@smillst I might have addressed your comments.  In places it was ambiguous whether a change was for style or for correctness.

**Review by smillst — APPROVED** [review]

**Pull request merged**